proxy/http2/Http2ConnectionState.cc

```diff
@@ -857,6 +857,7 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
 
   // Finalize HTTP/2 Connection
   case HTTP2_SESSION_EVENT_FINI: {
+    SCOPED_MUTEX_LOCK(lock, this->mutex, this_ethread());
```

Comment:
We saw some crashes if the ua stream mutex is not held.  OTOH, I am a bit confused, I don't see the mutex of Http2ConnectionState getting initialized. Most of the calls to the connection state are made while holding the ua_session's mutex